api/bucketsd/service.go

```diff
@@ -2349,14 +2355,14 @@ func (s *Service) Archive(ctx context.Context, req *pb.ArchiveRequest) (*pb.Arch
 		archiveConfig = &defaultDefaultArchiveConfig
 	}
 
+	if err := s.validateArchiveConfig(archiveConfig); err != nil {
+		return nil, fmt.Errorf("validating archive config: %s", err)
+	}
+
 	storageConfig := baseArchiveStorageConfig
 	storageConfig.Cold.Filecoin = toFilConfig(archiveConfig)
-
-	if storageConfig.Cold.Filecoin.Address == "" {
-		// We don't have an address to use, which is the case for a BucketArchive.DefaultArchiveConfig
-		// that is the default value, get the default address from the user
-		storageConfig.Cold.Filecoin.Address = defConfRes.DefaultStorageConfig.Cold.Filecoin.Address
-	}
+	// Get the address from the default storage config for this user.
+	storageConfig.Cold.Filecoin.Address = defConfRes.DefaultStorageConfig.Cold.Filecoin.Address
```

Comment:
We now always get the address from the user's default Powergate `StorageConfig`, they can't modify this via `ArchiveConfig` anymore.